lib/core/application/backendCluster.ts

```diff
@@ -0,0 +1,113 @@
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2015-2020 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import kerror from '../../kerror';
+import { JSONObject } from '../../../index';
+import { EventHandler } from '../../types';
+
+export class BackendCluster {
+  /**
+   * Broadcasts an event to listeners across all registered Kuzzle nodes
+   *
+   * @param  {string}        event
+   * @param  {JSONObject}    payload
+   * @return {Promise<void>}
+   */
+  async broadcast (event: string, payload: JSONObject): Promise<void> {
+    // @deprecated - to be removed with the new cluster (that event will
+    // always have an answerer)
+    if (!global.kuzzle.hasAskAnswerer('cluster:event:broadcast')) {
+      throw kerror.get('core', 'fatal', 'assertion_failed', 'Cannot use cluster methods in a single-node environment');
```

Comment:
Can't we have a simple return if Kuzzle is in single-node mode?

Otherwise developers need to check manually if Kuzzle is deployed as a cluster or not (and also there is no clean way to do this)

Reply:
This makes sense, since this doesn't change the addon behavior in any way.

I'll make that change right away.